recouple/src/type_rep.js

```diff
@@ -2,4 +2,30 @@
 export interface TypeRep<T> {}
 
 class StringRep implements TypeRep<string> {}
+class NumRep implements TypeRep<number> {}
+class NullRep implements TypeRep<null> {}
+class UndefinedRep implements TypeRep<typeof undefined> {}
+class UnionRep<A, B> implements TypeRep<A | B> {
+  left: A;
+  right: B;
+  constructor(a: A, b: B) {
+    this.left = a;
+    this.right = b;
+  }
+}
+
 export const string = new StringRep();
+export const number = new NumRep();
+export const nullValue = new NullRep();
```

Comment:
maybe we should export a named export called T, instead of exporting individually, so we can name our singletons `T.null`, and `T.undefined`, instead of `T.nullValue` and `T.undefinedValue`


Something like
```js
export const T = {
  null: new NullRep(),
  undefined: new UndefinedRep()
}
```